src/SFA.DAS.AAN.Hub.Data/Repositories/MemberDataCleanupRepository.cs

```diff
@@ -0,0 +1,121 @@
+    {
+        var statuses = new String[] { "Withdrawn", "Deleted" };
+        var query = _context.Members
+            .Where(m => statuses.Contains(m.Status) && m.Email != m.Id.ToString() &&
+                        m.EndDate < DateTime.Today.AddDays(-14))
+            .Include(m => m.MemberPreferences)
+            .Include(m => m.MemberProfiles)
+            .Include(m => m.Notifications)
+            .Include(m => m.Audits);
+
+        return await query.ToListAsync();
+    }
+
+    public async Task<List<Member>> GetRemovedMembers()
+    {
+        var query = _context.Members
+                .Where(m => m.Status == "Removed" && m.Email != m.Id.ToString())
+                .Include(m => m.MemberPreferences)
+                .Include(m => m.MemberProfiles)
+                .Include(m => m.Notifications)
+                .Include(m => m.Audits);
+
+        return await query.ToListAsync();
+    }
+
+    public async Task UpdateMemberDetails(Member member, CancellationToken cancellationToken)
+    {
+        var existingMember = await _context.Members.Where(m => m.Id == member.Id).FirstAsync(cancellationToken);
+
+        existingMember.FirstName = "Past";
+        existingMember.LastName = "Member";
+        existingMember.Email = member.Id.ToString();
+        existingMember.OrganisationName = "";
+        existingMember.IsRegionalChair = false;
+        existingMember.LastUpdatedDate = DateTime.UtcNow;
+    }
+
+    public void DeleteMemberProfile(List<MemberProfile> memberProfiles, CancellationToken cancellationToken)
+    {
+        _context.MemberProfiles.RemoveRange(memberProfiles);
+    }
+
+    public void DeleteMemberPreference(List<MemberPreference> memberPreferences, CancellationToken cancellationToken)
+    {
+        _context.MemberPreferences.RemoveRange(memberPreferences);
+    }
+
+    public void DeleteMemberNotifications(List<Notification> memberNotifications, CancellationToken cancellationToken)
+    {
+        _context.Notifications.RemoveRange(memberNotifications);
+    }
+
+    public void DeleteMemberAudit(List<Audit> memberAudits, CancellationToken cancellationToken)
+    {
+        var auditsToRemove = memberAudits.Select(a => a).Where(x => x.Resource != "Member").ToList();
+        _context.Audits.RemoveRange(auditsToRemove);
+    }
+
+    public async Task DeleteMemberApprentice(Member member, CancellationToken cancellationToken)
+    {
+        var apprentices = await _context.Apprentices.Where(a => a.MemberId == member.Id).FirstAsync(cancellationToken);
+
+        _context.Apprentices.RemoveRange(apprentices);
+    }
+
+    public async Task DeleteMemberEmployer(Member member, CancellationToken cancellationToken)
+    {
+        var employers = await _context.Employers.Where(a => a.MemberId == member.Id).FirstAsync(cancellationToken);
+
+        _context.Employers.RemoveRange(employers);
+    }
+
+    public async Task UpdateMemberFutureAttendance(Member member, CancellationToken cancellationToken)
```

Comment:
`principle of least knowledge` you only need member id here then why pass in a full blown Member entity

Reply:
I would move this logic out of repository, as repositories are not unit testable, they should be dumb , you give them a  list of entities to remove, it should just remove them.

Reply:
Logic moved to the service and unit tested